core/invoker/src/main/scala/whisk/core/containerpool/ContainerProxy.scala

```diff
@@ -419,9 +419,10 @@ class ContainerProxy(
 
     // Disambiguate activation errors and transform the Either into a failed/successful Future respectively.
     activationWithLogs.flatMap {
-      case Right(act) if !act.response.isSuccess => Future.failed(ActivationUnsuccessfulError(act))
-      case Left(error)                           => Future.failed(error)
-      case Right(act)                            => Future.successful(act)
+      case Right(act) if !act.response.isSuccess && !act.response.isApplicationError =>
```

Comment:
@tysonnorris, is it possible to just check for `!act.response.isApplicationError` here instead?